pkg/termination/termination_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package termination
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/logging"
+)
+
+func TestExistingFile(t *testing.T) {
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "tempFile")
+	if err != nil {
+		log.Fatal("Cannot create temporary file", err)
+	}
+	// Remember to clean up the file afterwards
+	defer os.Remove(tmpFile.Name())
+
+	logger, _ := logging.NewLogger("", "test termination")
+	defer logger.Sync()
+	output := []v1alpha1.PipelineResourceResult{}
+	now := time.Now()
+	output = append(output, v1alpha1.PipelineResourceResult{
+		Key:   "StartedAt",
+		Value: now.Format(time.RFC3339),
+	})
+
+	WriteMessage(logger, tmpFile.Name(), output)
+
+	output = []v1alpha1.PipelineResourceResult{}
+	now = time.Now()
+	output = append(output, v1alpha1.PipelineResourceResult{
+		Key:   "FinishedAt",
+		Value: now.Format(time.RFC3339),
+	})
+
+	WriteMessage(logger, tmpFile.Name(), output)
+
+	exists := Exists(tmpFile.Name())
+	if err != nil {
+		logger.Fatalf("Error checking file existance %v: %v", tmpFile.Name(), err)
+	}
+	if exists {
+		fileContents, err := ioutil.ReadFile(tmpFile.Name())
+		if err != nil {
+			logger.Fatalf("Unexpected error reading %v: %v", tmpFile.Name(), err)
+		}
+		if err != nil {
+			logger.Fatalf("temp file cannot be deleted\n")
+		}
+		var existingEntries []v1alpha1.PipelineResourceResult
+
+		if err := json.Unmarshal([]byte(fileContents), &existingEntries); err != nil {
+			logger.Fatalf("Could not parse json message '%q' because of %w\n", fileContents, err)
+		}
+		if len(existingEntries) != 2 {
```

Comment:
Let's compare the exact expected contents, using `cmp.Diff` -- there should be other examples using `cmp.Diff` strewn about, but basically:

```go
want := `{"StartedAt":"...","FinishedAt":"..."}`
b, err := ioutil.ReadFile(tmpFile)
if d := cmp.Diff(want, string(b)); d != "" {
  t.Fatalf("Diff(-want, got): %s", d)
}
```

This means we should probably not have dynamic `time.Now()` values written above -- we can just add a value that says `"hello"` and see that it gets appended.

Reply:
done